### ISSUE TITLE
HDF read fails if creation order is not tracked

### ISSUE THREAD

**Issue by couletj** [CGNS/CGNS#694]
Some functions of `ADFH.c` uses an iteration pattern based on the creation order of the hdf groups, see eg:

```C
#if H5_VERSION_GE(1,12,0)
    H5Literate2(gid, H5_INDEX_CRT_ORDER, H5_ITER_NATIVE, NULL, fix_dimensions, NULL);
#else
    H5Literate(gid, H5_INDEX_CRT_ORDER, H5_ITER_NATIVE, NULL, fix_dimensions, NULL);
#endif 
```

It seems that these iterations ignore the groups which have been created without the CRT_ORDER_TRACKED property flag.

First, I am wondering what is the expected behaviour of the HDF routine [H5Literate2](https://docs.hdfgroup.org/hdf5/develop/group___t_r_a_v.html#title10) : 

> idx_type specifies the index to be used. If the links have not been indexed by the index type, they will first be sorted by that index then the iteration will begin; if the links have been so indexed, the sorting step will be unnecessary, so the iteration may begin more quickly.

What I understand here is that there is a distinction if "ORDER_INDEXED" flag have been used, but nothing is said on what appends if "ORDER_TRACKED" is disabled.
 

If the behaviour of HDF5 lib is correct, when sould either

- state somewhere that the tracking of creation order is required
- or use the H5_INDEX_NAME iteration order

**Comment by couletj** [comment]
By the way, does anyone know the tool to use to see if creation order is or not tracked for a given hdf file ? hdfview reports this information but I did not find the equivalent command line tool to use.

**Comment by MicK7** [comment]
From my understanding there was a change in CGNS MLL implementation around 3.x release requiring to have INDEX_CRT_ORDER by default. Thus recent hdf5 file should not use the INDEX_NAME.
But, in develop version of CGNS/MLL a change was recently added to detected if the whole file use CRT_ORDER or not and then read the old 2.5 files. It is possible that some bug or incompability are still present.

What is the usecase that use H5_INDEX_NAME ?

@brtnfld   any comments ?

**Comment by brtnfld** [comment]
You are correct; this oversight was fixed in PR #690 to develop.

Can you try it with develop and see if you still have an issue? I might have missed an H5Literate correction.

As far as I can tell, there's no hdf5 tool that displays the creation order.

**Comment by couletj** [comment]
@brtnfld  I just built the library using develop branch and the problem is gone. Thanks for the PR !

@MicK7 We used to write hdf files using our own tools instead of MLL implementation, that is why we have name indexed files.
We are going to change these tools to save the CRT_ORDER, but as I said above maybe this should be stated somewhere in the docs ? I also add a similar issue concerning misleading documentation few mounths ago : https://github.com/CGNS/CGNS/issues/678.
